packages/rum-core/src/domain/trackPageActivities.ts

```diff
@@ -66,6 +67,8 @@ export function trackPageActivities(
   let firstRequestIndex: undefined | number
   let pendingRequestsCount = 0
 
+  const DOMMutationCollection = startDOMMutationCollection(lifeCycle)
```

Comment:
two design concerns with this approach:

- using lifeCycle does not seem to be needed anymore since the component that needs the mutation is the one that start the mutation observer
- if another component need to use a mutation observer, it will use the same instance but stop conditions could be different

Reply:
In fact, we can already have several `waitIdlePageActivity` at the same time so stopping the shared mutation observer instance can already be an issue

Reply:
A new instance of `MutationObserver` with its own `stop()` is created each time we call `startDOMMutationCollection` function. So we can have multiple call to `waitIdlePageActivity` without issues or maybe I miss something?

Reply:
oh my bad, I thought it was a singleton but it is not. 
So then we can have several MutationObservers in parallel, meaning that one dom mutation will trigger several DOM_MUTATED lifecycle events 🤔
It could be nice to instead have a single observer and keep track of subscribers to stop it when nobody needs it anymore.


what about having something like:
```ts
export function trackPageActivities(
  lifeCycle: LifeCycle,
  mutations: Mutations
) {
  
  ...

  subscriptions.push(mutations.subscribe(() => notifyPageActivity()))

  ...

  return {
    stop: () => {
      subscriptions.forEach((s) => s.unsubscribe())
    }
  }
}
```
mutations could have a similar API than lifeCycle limiting the impact to the tests